lib/Doctrine/ODM/PHPCR/DocumentManager.php

```diff
@@ -477,13 +477,23 @@ public function getDocumentsByPhpcrQuery(QueryInterface $query, $className = nul
 
         $result = $query->execute();
 
-        $ids = array();
+        $documents = array();
         foreach ($result->getRows() as $row) {
             /** @var $row \PHPCR\Query\RowInterface */
-            $ids[] = $row->getPath($primarySelector);
+            $id = $row->getPath($primarySelector);
+            if (!isset($documents[$id])) {
+                // TODO: we have to fallback here or TranslationConverter fails, but im not sure if that's a
+                //  bug in the TranslationConverter that only now comes to the surface
+                $hints = array('fallback' => true);
+                $documents[$id] = $this->getUnitOfWork()->getOrCreateDocument(
+                    $className,
+                    $row->getNode($primarySelector),
+                    $hints
+                );
+            }
         }
 
-        return $this->findMany($className, $ids);
```

Comment:
if the problem is duplicate ids, could we `array_unique($ids)` somewhere? findMany does getOrCreateDocuments with the list of nodes - as far as i can see, getOrCreateDocuments does not check if we have duplicates, but does some refresh if asked to, and other funny stuff (ouch, that method is complicated). imho it would make sense to array_unique the ids at the start of findMany, and then build a hashmap of documents and in the end loop over the original ids list to re-create the collection in the right order and with duplicates if ids have been repeated. also, getOrCreateDocuments will kill duplicates because $documents is a hashmap with the node id being the key...

Reply:
oh, actually. if i read getOrCreateDocuments correctly, we return a hashmap anyways, so to keep the same behaviour as before, we can simply array_unique in findMany, as duplicates already are written to the same array field and therefore we only get each document once anyways, right?

Reply:
I also think adding array_unique to `findMany` is indeed the solution we can provide for doctrine/phpcr-odm.

I do not think we can return duplicates if duplicates are asked as that would break the behaviour used for `getDocumentsByPhpcrQuery`. I'll make the PR and target both branches including a test

Reply:
great, thanks!

yeah returning duplicates would also be multiple BC breaks for findMany, because a) it did not return duplicates until now and b) the index are the node ids, and we would need to change what ids we use to even allow returning duplicates :-O